### PR TITLE
Ensure Clean State of Nodes After Commit in Besu-Verkle-Trie

### PR DESCRIPTION
**Pull request by neotheprogramist**
fixes https://github.com/neotheprogramist/besu-verkle-trie/issues/1
fixes https://github.com/hyperledger/besu-verkle-trie/issues/15

### CODE REVIEW

**Review by thomas-quadratic — COMMENTED** [review]
Thanks, seems that the logic is correct.
The tests are quite light though as it is, testing only single put, commit and check trie is clean.
Could you please add tests that would do multiple put, then commit, then put some more elements, and check that only the affected nodes are dirty, the others remain clean ?

File: src/test/java/org/hyperledger/besu/ethereum/trie/verkle/NodeDirtyTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  @Test
+  public void testOneValueSimpleVerkleTrie() {
+    NodeUpdaterMock nodeUpdater = new NodeUpdaterMock();
+    SimpleVerkleTrie<Bytes32, Bytes32> trie = new SimpleVerkleTrie<>();
+    Bytes32 key =
+        Bytes32.fromHexString("0x00112233445566778899aabbccddeeff00112233445566778899aabbccddeeff");
+    Bytes32 value =
+        Bytes32.fromHexString("0x1000000000000000000000000000000000000000000000000000000000000000");
+    trie.put(key, value);
+
```
Comment (matkt) [inline src/test/java/org/hyperledger/besu/ethereum/trie/verkle/NodeDirtyTest.java:36]: could we add an assert do check that isdirty is true before the commit and after the put ?

File: src/test/java/org/hyperledger/besu/ethereum/trie/verkle/NodeDirtyTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    NodeLoaderMock nodeLoader = new NodeLoaderMock(nodeUpdater.storage);
+    StoredNodeFactory<Bytes32> nodeFactory =
+        new StoredNodeFactory<>(nodeLoader, value -> (Bytes32) value);
+    StoredVerkleTrie<Bytes32, Bytes32> trie = new StoredVerkleTrie<>(nodeFactory);
+    Bytes32 key =
+        Bytes32.fromHexString("0x00112233445566778899aabbccddeeff00112233445566778899aabbccddeeff");
+    Bytes32 value =
+        Bytes32.fromHexString("0x1000000000000000000000000000000000000000000000000000000000000000");
+    trie.put(key, value);
+
```
Comment (matkt) [inline src/test/java/org/hyperledger/besu/ethereum/trie/verkle/NodeDirtyTest.java:53]: could we add an assert do check that isdirty is true before the commit and after the put ?

File: src/main/java/org/hyperledger/besu/ethereum/trie/verkle/node/LeafNode.java [hunk 3]
```diff
@@ -50,6 +50,7 @@ public LeafNode(final Bytes location, final V value) {
     this.value = value;
     this.valueSerializer = val -> (Bytes) val;
   }
+
```
Comment (matkt) [inline src/main/java/org/hyperledger/besu/ethereum/trie/verkle/node/LeafNode.java:53]: we can remove this empty line update

**Review by matkt — APPROVED** [review]
Some minors change to do but after this modification we will be able to merge. Tried and it's working fine @thomas-quadratic

**Pull request merged**